compiler/semstmts.nim

```diff
@@ -445,55 +449,114 @@ proc setVarType(c: PContext; v: PSym, typ: PType) =
         "; new type is: " & typeToString(typ, preferDesc))
   v.typ = typ
 
-proc semLowerLetVarCustomPragma(c: PContext, a: PNode, n: PNode): PNode =
-  var b = a[0]
+proc copyExcept(n: PNode, i: int): PNode =
+  result = copyNode(n)
+  for j in 0..<n.len:
+    if j != i: result.add(n[j])
+
+proc semVarCustomPragma(c: PContext, a: PNode, n: PNode): PNode =
+  result = nil
+  if a.len > 3:
+    # a, b {.prag.}: int = 3 not allowed
+    return
```

Comment:
Too many `return` and `continue` statements, obscuring all the loop invariants. Use structured programming and more helper procs.